### DISCUSSION TITLE
problem running the notebook: partition_based_clustering.ipynb

### DISCUSSION THREAD

**Discussion by drakeleeb** [alan-turing-institute/sktime#1898] [Q&A] [answered]
Here are the errors:
Executing this code:
from sklearn.model_selection import train_test_split
from sktime.clustering._k_means import TimeSeriesKMeans
from sktime.clustering._k_medoids import TimeSeriesKMedoids
from sktime.clustering.utils.plotting._plot_partitions import plot_cluster_algorithm
from sktime.datasets import load_arrow_head

Results in these errors:
ModuleNotFoundError                       Traceback (most recent call last)
/var/folders/24/msxkh3ks0lbcr0cqhcrpg53r8kl7p_/T/ipykernel_91572/3417722633.py in <module>
      3 from sktime.clustering._k_means import TimeSeriesKMeans
      4 from sktime.clustering._k_medoids import TimeSeriesKMedoids
----> 5 from sktime.clustering.utils.plotting._plot_partitions import plot_cluster_algorithm
      6 from sktime.datasets import load_arrow_head

ModuleNotFoundError: No module named 'sktime.clustering.utils'
----------------------

I tried reinstalling the sktime package several times and even reinstalled Anaconda. Nothing
has worked so far. I see in the sktime repo that the code exists. And, in fact, I was running it
successfully until resent updates. I've checked the pkgs directory in my Python installation and, 
indeed, sktime.clustering.utils is not in the directory tree.

Thanks,
Barry

**Comment by drakeleeb** [comment]
The code is in:
from sktime.clustering.evaluation.plotting._plot_partitions import plot_cluster_algorithm
not in
from sktime.clustering.utils.plotting._plot_partitions import plot_cluster_algorithm
as in the current notebook.

In my site_packages directory it is in the path with evaluations not utils.

**Comment by fkiraly** [comment]
FYI @chrisholder, @TonyBagnall

**Comment by chrisholder** [comment, marked as answer]
Hi @drakeleeb good spot forgot to update this we recently refactored our clustering stuff! Ill push a fix to this!

**Comment by drakeleeb** [comment]
Hello Chris, Thanks! I’ll check out the changes. - Barry

**Comment by fkiraly** [comment]
@chrisholder, @drakeleeb, is this fixed/addressed?

**Comment by drakeleeb** [comment]
Fixed! Thank you. - Barry
